libkbfs/errors.go

```diff
@@ -165,18 +165,32 @@ func (e ReadAccessError) Error() string {
 		e.User, buildCanonicalPath(e.Public, e.Tlf))
 }
 
-// WriteAccessError indicates that the user tried to read from a
-// top-level folder without read permission.
+// WriteAccessError indicates an error when trying to write a file
 type WriteAccessError struct {
-	User   libkb.NormalizedUsername
-	Tlf    CanonicalTlfName
-	Public bool
+	User     libkb.NormalizedUsername
+	Filename string
+	Tlf      CanonicalTlfName
+	Public   bool
 }
 
 // Error implements the error interface for WriteAccessError
 func (e WriteAccessError) Error() string {
-	return fmt.Sprintf("%s does not have write access to directory %s",
-		e.User, buildCanonicalPath(e.Public, e.Tlf))
+	if e.Tlf != "" {
+		return fmt.Sprintf("%s does not have write access to directory %s",
+			e.User, buildCanonicalPath(e.Public, e.Tlf))
+	}
+	return fmt.Sprintf("%s does not have write access to %s", e.User, e.Filename)
+}
+
+// WriteUnsupportedError indicates an error when trying to write a file
+type WriteUnsupportedError struct {
+	User     libkb.NormalizedUsername
+	Filename string
+}
+
+// Error implements the error interface for WriteAccessError
+func (e WriteUnsupportedError) Error() string {
+	return fmt.Sprintf("%s can't write to %s", e.User, e.Filename)
```

Comment:
I don't think we need the username in here, since it's not supported for anyone.  Maybe something like simple like `("Writing to %s is unsupported", e.Filename)`, and get rid of the need for `User` (and the `GetCurrentUserInfo` call in `writeUnsupportedError`).